### ISSUE TITLE
BUG: eFiled document has duplicate Docket entry #'s, no cover sheet, and incorrect page count

### ISSUE THREAD

**Issue by ttlenard** [ustaxcourt/ef-cms#8076]
**Describe the Bug**

There are two ustaxcourt/ef-cms-flexion#14 index numbers for case at docket number 3517-21.  The first one has "0" pages beside it, and there is no cover page in the first ustaxcourt/ef-cms-flexion#14.  The second ustaxcourt/ef-cms-flexion#14 is ok.  This was a motion to consolidate cases.  The motion filed in 3521-21 is okay.  This document was eFiled by the petitioner.  

It seems that this is a re-occurring bug?  I found very similar issues reported in: [8711](https://github.com/flexion/ef-cms/issues/8711), [7487](https://github.com/flexion/ef-cms/issues/7487), and [6777](https://github.com/flexion/ef-cms/issues/6777).

Note: I did try to look in the logs to see if I could find anything...nothing jumped out at me, but I could be missing something.  It does appear that when the IRS respondent filed both motions to consolidate, both were successful.
https://search-info-4vkmt6q7gikrqmth5bknb4mj6u.us-east-1.es.amazonaws.com/_plugin/kibana/app/discover#/?_g=(filters:!(),refreshInterval:(pause:!t,value:0),time:(from:'2021-09-13T14:00:00.000Z',to:'2021-09-14T00:00:00.000Z'))&_a=(columns:!(_source),filters:!(('$state':(store:appState),meta:(alias:!n,disabled:!f,index:f8384e80-2516-11eb-82a4-0f6df8788ac7,key:request.method,negate:!t,params:(query:GET),type:phrase),query:(match_phrase:(request.method:GET)))),index:f8384e80-2516-11eb-82a4-0f6df8788ac7,interval:auto,query:(language:kuery,query:'3517'),sort:!())

**Business Impact/Reason for Severity**
high severity

**In which environment did you see this bug?**
Prod

**Who were you logged in as?**
Docket 

**What were you doing when you discovered this bug? (Using the application, demoing, smoke tests, testing other functionality, etc.)**
Using the application

**To Reproduce**

**Expected Behavior**
All eFiled documents should have a cover sheet, and the docket record should correctly indicate how many pages are in the document.  There should not be duplicate docket entry #'s.
If a cover sheet can't be applied, the PDF should be rejected, not uploaded, and the user would see the 400 error

**Actual Behavior**
A clear and concise description of what actually happened.
There are two docket entry ustaxcourt/ef-cms-flexion#14's, One is missing the cover sheet and is showing 0 pages.  

**Screenshots**
![image.png](https://images.zenhubusercontent.com/60f99b8ac0d8d04c8aed1aa3/92924d8a-643f-49e8-a20f-3212a2e62d05)

**Desktop (please complete the following information):**
 - Browser [e.g. chrome, safari] Edge

**Smartphone (please complete the following information):**

**Cause of Bug, If Known**

**Process for Logging a Bug:**
* Complete the above information
* Add a severity tag (Critical, High Severity, Medium Severity or Low Severity). See below for priority definition. 

**Severity Definition:**
* Critical Defect
Blocks entire system's or module’s functionality
No workarounds available
Testing cannot proceed further without bug being fixed.

* High-severity Defect
Affects key functionality of an application
There's a workaround, but not obvious or easy
App behaves in a way that is strongly different from the one stated in the requirements

* Medium-severity Defect
A minor function does not behave in a way stated in the requirements.
Workaround is available and easy

* Low-severity Defect
Mostly related to an application’s UI
Doesn't need a workaround, because it doesn't impact functionality

## Definition of Done (Updated 4-14-21)
**Product Owner**
 - [ ]  Bug fix has been validated in the Court's test environment

**Engineering**
 - [ ] Automated test scripts have been written
 - [ ] Field level and page level validation errors (front-end and server-side) integrated and functioning
 - [ ] Verify that language for docket record for internal users and external users is identical
 - [ ] New screens have been added to pa11y scripts
 - [ ] All new functionality verified to work with keyboard and macOS voiceover https://www.apple.com/voiceover/info/guide/_1124.html 
 - [ ] READMEs, other appropriate docs, JSDocs and swagger/APIs fully updated
 - [ ] UI should be touch optimized and responsive for external only (functions on supported mobile devices and optimized for screen sizes as required)
 - [ ] Interactors should validate entities before calling persistence methods
 - [ ] Code refactored for clarity and to remove any known technical debt
 - [ ] Deployed to the Court's test environment

**Comment by rosiehsmith** [comment]
@ttlenard just to make sure we're on the right track to reproduce this bug (on mig) - this is what we've tried so far:

1. login as a petitioner on case 3521-21 
2. go to case 3521-21
3. file a motion to consolidate cases as the petitioner 
4. login as docket clerk 
5. mark QC complete 
6. view docket record for 3521-21 (no duplicates)
7.  view docket record for 3517-21 (no duplicates)

do you have a suggestion for something else we should try?

for context https://app.mig.ef-cms.ustaxcourt.gov/case-detail/3517-21

**Comment by ttlenard** [comment]
@rosiehsmith I think that the test case you provided is mostly correct, but I have a few suggestions.  I added in the other case that the petitioner is filing a motion for.  I would think that the 

- login as a petitioner on case 3521-21
- go to case 3521-21
- file a motion to consolidate cases as the petitioner
- ensure that the motion has a cover sheet and page number, ensure no duplicate docket entry
- go to case 3517-21
- file a motion to consolidate cases as the petitioner
- Ensure that the motion has a cover sheet and page number, ensure no duplicate docket entry
- login as docket clerk
- mark QC complete
- view docket record for 3521-21 (no duplicates)
- view docket record for 3517-21 (no duplicates)

**Comment by mariahkannenberg** [comment]
Try a few steps to reproduce (Tenille's new steps). If unable, run a gluejob in MIG so that we can attempt to reproduce.  T

**Comment by rosiehsmith** [comment]
Unfortunately I was still unable to reproduce with the above steps (on mig)

**Comment by mariahkannenberg** [comment]
Blocked - Let's try a gluejob @mmarcotte @cholly75 @adunkman

**Comment by rosiehsmith** [comment]
moving this back to the backlog to prioritize currently working on 8511 but per @mmarcotte it's recommended to attempt reproducing on the `test` env (mentioned in thread [here](https://flexion.slack.com/archives/CD4PNKEPK/p1632498579128100))

**Comment by ttlenard** [comment]
I looked in TEST environment and for this Docket record, there are 2 docket entry # 14's.  We don't need a Glue Job since the data is there in TEST already.

**Comment by matthopson** [comment]
I've also tried to reproduce various ways. The best explanation I can come up with, based on the payload of the request(s), within 10 seconds of filing the document, _something_ went wrong - perhaps on the client's network - that allowed the document to continue to be uploaded / saved to persistence, but also allowed the user to go back and re-submit before the case was updated with the previously filed document. When the document was re-submitted, the other document wasn't yet on the docket record, so it wasn't accounted for when generating the index.

I think we just need to remove the first submission directly as it seems to be the one that didn't get processed correctly.
